src/transformState.js

```diff
@@ -5,7 +5,23 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const obj of actions) {
+    if (obj.type === 'addProperties') {
+      Object.assign(state, obj.extraData);
+    } else if (obj.type === 'removeProperties') {
+      for (const el of obj.keysToRemove) {
+        if (state.hasOwnProperty(el)) {
+          delete state[el];
+        }
+      }
+    } else if (obj.type === 'clear') {
+      for (const deletedItem in state) {
+        delete state[deletedItem];
+      }
+    }
```

Comment:
Looks complicated, that's a great place to use switch/case

Reply:
Fixed